src/models/user.ts

```diff
@@ -105,11 +105,10 @@ UserSchema.statics.validatePassword = async (
   return user;
 };
 
-UserSchema.statics.findByCredentials = async (
-  email: string,
-  password: string,
-) => {
-  const user = await UserModel.findOne({ email });
+UserSchema.statics.findByCredentials = async (id: string, password: string) => {
+  const userWithEmail = await UserModel.findOne({ email: id });
+  const userWithWatIAM = await UserModel.findOne({ watIAMUserId: id });
+  const user = userWithEmail ? userWithEmail : userWithWatIAM;
```

Comment:
We can merge these two DB requests with the `$or` operator.

Also, this would've been a good place to use the `??` operator but it won't be necessary if we merge the DB requests

Reply:
I kind of bricked this PR by merging the other 2 in prematurely. However I did update that code here: 782323bff3f0fdf32eefaa498192cc945432b3b1